spi_nand_flash/test_app/main/test_spi_nand_flash.c

```diff
@@ -210,3 +211,76 @@ TEST_CASE("copy nand flash sectors", "[spi_nand_flash]")
     }
     deinit_nand_flash(nand_flash_device_handle, spi);
 }
+
+TEST_CASE("verify mark_bad_block works", "[spi_nand_flash]")
+{
+    spi_nand_flash_device_t *nand_flash_device_handle;
+    spi_device_handle_t spi;
+    setup_nand_flash(&nand_flash_device_handle, &spi);
```

Comment:
As far as I understand, setup_nand_flash also initializes Dhara. Since the rest of the test case verifies lower-level APIs, it would make sense to not initialize Dhara and only call `nand_` (rather than `spi_nand_flash_`) prefixed APIs.

Reply:
I see, I'm okay merging this as is, we can refactor this in a follow-up MR.